### PR TITLE
[Rule Tuning] Suspicious Explorer Child Process

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/1034

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/initial_access_via_explorer_suspicious_child_parent_args.toml [hunk 1]
```diff
@@ -23,9 +23,17 @@ type = "eql"
 
 query = '''
 process where event.type in ("start", "process_started") and
-  process.name : ("cscript.exe", "wscript.exe", "powershell.exe", "rundll32.exe", "cmd.exe", "mshta.exe", "regsvr32.exe") and
+  (
+   process.name : ("cscript.exe", "wscript.exe", "powershell.exe", "rundll32.exe", "cmd.exe", "mshta.exe", "regsvr32.exe") or
+   process.pe.original_file_name : ("cscript.exe", "wscript.exe", "powershell.exe", "rundll32.exe", "cmd.exe", "mshta.exe", "regsvr32.exe")
```
Comment (brokensound77) [inline rules/windows/initial_access_via_explorer_suspicious_child_parent_args.toml:28]:
```suggestion
   process.pe.original_file_name in ("cscript.exe", "wscript.exe", "powershell.exe", "rundll32.exe", "cmd.exe", "mshta.exe", "regsvr32.exe")
```

(verify casing on these)

File: rules/windows/initial_access_via_explorer_suspicious_child_parent_args.toml [hunk 2]
```diff
@@ -23,9 +23,17 @@ type = "eql"
...
+  ) and
   /* Explorer started via DCOM */
-  process.parent.name : "explorer.exe" and process.parent.args : "-Embedding"
+  process.parent.name : "explorer.exe" and process.parent.args : "-Embedding" and
+  not process.parent.args:
+          (
+            "/factory,{5BD95610-9434-43C2-886C-57852CC8A120}",
+            "/factory,{ceff45ee-c862-41de-aee2-a022c81eda92}"
```
Comment (brokensound77) [inline rules/windows/initial_access_via_explorer_suspicious_child_parent_args.toml:35]: can you add a comment on what these are
```suggestion
            /* <explanation> */
            "/factory,{5BD95610-9434-43C2-886C-57852CC8A120}",
            "/factory,{ceff45ee-c862-41de-aee2-a022c81eda92}"
```

File: rules/windows/initial_access_via_explorer_suspicious_child_parent_args.toml (hunk 1 above)
Comment (Samirbous) [reply rules/windows/initial_access_via_explorer_suspicious_child_parent_args.toml:28]: its ok for this instance but if more than 10 processes verifying each become time consuming.

**Review by threat-punter — APPROVED** [review]

File: rules/windows/initial_access_via_explorer_suspicious_child_parent_args.toml [hunk 3]
```diff
@@ -23,9 +23,18 @@ type = "eql"
...
+   process.pe.original_file_name in ("cscript.exe", "wscript.exe", "PowerShell.EXE", "RUNDLL32.EXE", "Cmd.Exe", "MSHTA.EXE", "REGSVR32.EXE")
+  ) and
   /* Explorer started via DCOM */
-  process.parent.name : "explorer.exe" and process.parent.args : "-Embedding"
+  process.parent.name : "explorer.exe" and process.parent.args : "-Embedding" and
+  not process.parent.args:
+          (
+            /* Noisy CLSID_SeparateSingleProcessExplorerHost Explorer COM Class IDs   */
+            "/factory,{5BD95610-9434-43C2-886C-57852CC8A120}",
+            "/factory,{ceff45ee-c862-41de-aee2-a022c81eda92}"
```
Comment (bm11100) [inline rules/windows/initial_access_via_explorer_suspicious_child_parent_args.toml:36]: looks like all other noise is coming from `/factory,{75dff2b7-6936-4c06-a8bb-676a7b00b24b}`

Comment (Samirbous) [reply rules/windows/initial_access_via_explorer_suspicious_child_parent_args.toml:36]: that clsid is always invoked when malware use explorer.exe as exec trampoline ("explorer.exe C:\path\script.vbs" automatically windows will spawn a separate explorer.exe with that cmdline (75df..) and spawn  the right script extension handler (e.g. wscript.exe for .vbs) we don't want to miss that (if we receive more tuning issues we can exclude more processes).

**Review by bm11100 — APPROVED** [review]

**Pull request merged**